vegatime/service.go

```diff
@@ -80,6 +90,8 @@ func (s *Svc) GetTimeLastBatch() time.Time {
 func (s *Svc) notify(ctx context.Context, t time.Time) {
 	s.mu.Lock()
```

Comment:
If we're locking here *and* in `GetTimeNow()` then any of the listener callbacks that call `GetTimeNow()` are going to deadlock. Maybe a read lock will fix all? Also do we need to now lock in `SetTimeNow()`?

I guess this is what the previous approach avoided, complex locking....

Reply:
@wwestgarth , I missed that sorry.
Yes, RWMutex exclusion lock on the service can be used to RLock `GetTimeNow` and `notify` procedures.

`SetTimeNow` is currently used by stubs in integration, as well as for testing in mocks only except here.
So it would be safe to call `notify` with a RLock in it I think.

Reply:
why do we need to lock here at all?